tests/test_load_azd_env.py

```diff
@@ -29,53 +30,55 @@ def _azd_env_set(key: str, value: str, *, cwd: Path) -> str:
 
 
 def test_load_azd_env(tmp_path: Path) -> None:
-    from os import getenv
 
     with open(tmp_path / "azure.yaml", "w") as config:
         config.write("name: dotenv-azd-test\n")
 
     _azd_env_new("MY_AZD_ENV", cwd=tmp_path)
     var_set = load_azd_env(cwd=tmp_path)
-    assert getenv("AZURE_ENV_NAME") == "MY_AZD_ENV"
+    assert os.getenv("AZURE_ENV_NAME") == "MY_AZD_ENV"
     assert var_set
 
 
-def test_load_azd_env_override(tmp_path: Path) -> None:
-    from os import environ, getenv
+def test_load_azd_env_override(tmp_path: Path, monkeypatch) -> None:
 
     with open(tmp_path / "azure.yaml", "w") as config:
         config.write("name: dotenv-azd-test\n")
 
-    environ["VAR1"] = "INITIAL"
+    monkeypatch.setenv("VAR1", "INITIAL")
     _azd_env_new("MY_AZD_ENV", cwd=tmp_path)
     _azd_env_set("VAR1", "OVERRIDE", cwd=tmp_path)
     var_set = load_azd_env(cwd=tmp_path)
-    assert getenv("AZURE_ENV_NAME") == "MY_AZD_ENV"
-    assert getenv("VAR1") == "INITIAL"
+    assert os.getenv("AZURE_ENV_NAME") == "MY_AZD_ENV"
+    assert os.getenv("VAR1") == "INITIAL"
     assert var_set
     var_set = load_azd_env(cwd=tmp_path, override=True)
-    assert getenv("VAR1") == "OVERRIDE"
+    assert os.getenv("VAR1") == "OVERRIDE"
     assert var_set
 
 
 def test_load_azd_env_no_project_exists_error(tmp_path: Path) -> None:
-    from dotenv_azd import AzdNoProjectExistsError
-
     with pytest.raises(AzdNoProjectExistsError, match="no project exists"):
         load_azd_env(cwd=tmp_path)
 
 
-def test_load_azd_env_azd_command_not_found_error(tmp_path: Path) -> None:
-    from os import environ
-
-    from dotenv_azd import AzdCommandNotFoundError
-
-    path = environ["PATH"]
-    environ["PATH"] = ""
+def test_load_azd_env_azd_command_not_found_error(tmp_path: Path, monkeypatch) -> None:
+    monkeypatch.setenv("PATH", "")
     with pytest.raises(AzdCommandNotFoundError):
         load_azd_env(cwd=tmp_path)
-    environ["PATH"] = path
```

Comment:
it's not needed to restore the PATH before the method exits? is `monkeypatch.setenv` taking care of this?

Reply:
Sounds good